crates/solvers/src/domain/solution.rs

```diff
@@ -4,12 +4,31 @@ use {
         util,
     },
     ethereum_types::{Address, U256},
-    std::{collections::HashMap, slice},
+    std::{
+        collections::HashMap,
+        slice,
+        sync::atomic::{AtomicU64, Ordering},
+    },
 };
 
+static COUNTER: AtomicU64 = AtomicU64::new(1);
+pub fn solution_id() -> Id {
```

Comment:
nit: Now you'd have to use this as `solution::solution_id()`. Maybe it's slightly nicer to read to have it as a member on `Id` like `Id::generate()` or `Id::next()`.

Reply:
Yes, added.